index.ts

```diff
@@ -3,7 +3,18 @@ import mapAgeCleaner from 'map-age-cleaner';
 
 type AnyFunction = (...arguments_: any) => any;
 
-const decoratorInstanceMap = new WeakMap();
+/**
+Two dimensional WeakMap, mapping the combination of an _instance_ and a
+_propertyDescriptor_ to a memoized function.
+
+The memoized function that belongs to a specific instance and a specific
+propertyDescriptor could theoretically be accessed as follows:
+
+```
+const mem = decoratorInstanceMap.get(instance)?.get(propertyDescriptor)
+```
+ */
+const decoratorInstanceMap = new WeakMap<object, WeakMap<PropertyDescriptor, unknown>>(); // eslint-disable-line @typescript-eslint/ban-types
```

Comment:
I think the solution is to just move the weak map to the decorator function, but I'm not sure.

Reply:
That's actually a pretty good idea! Tested it, works!